acl.go

```diff
@@ -0,0 +1,104 @@
+package imap
+
+import (
+	"strings"
+)
+
+// IMAP4 ACL extension (RFC 2086)
+
+type RightSet string
```

Comment:
It's a bit too tempting for users to try to compare `RightSet` with `==` (which compiles because they're strings, but doesn't do the right thing). Maybe we should turn this into a `[]Right`?